spec/jobs/zipmaker_job_spec.rb

```diff
@@ -3,13 +3,77 @@
 describe ZipmakerJob, type: :job do
   let(:druid) { 'bj102hs9687' }
   let(:version) { 1 }
+  let(:binary_path) { Moab::StorageServices.object_version_path(druid, version) }
+  let(:zip_path) { "spec/fixtures/transfers/bj/102/hs/9687/#{druid}#{format('.v%04d.zip', version)}" }
+
+  before do
+    allow(PlexerJob).to receive(:perform_later).with(any_args)
+    allow(Settings).to receive(:zip_storage).and_return('spec/fixtures/transfers')
+  end
 
   it 'descends from ApplicationJob' do
     expect(described_class.new).to be_an(ApplicationJob)
   end
 
-  it 'invokes PlexerJob' do
-    expect(PlexerJob).to receive(:perform_later).with(druid, version)
-    described_class.perform_now(druid, version)
+  context 'zip already exists in zip storage' do
+
+    it 'does nothing' do
+      expect(File).to exist(zip_path)
+      expect(described_class).not_to receive(:zip_binary)
+      described_class.perform_now(druid, version)
+    end
+  end
+
+  context 'zip is not yet in zip storage' do
+    let(:version) { 2 }
+
+    after { File.delete(zip_path) }
+
+    it 'zips up the druid version into zip storage' do
+      described_class.perform_now(druid, version)
+      expect(File).to exist(zip_path)
+    end
+
+    it 'invokes PlexerJob' do
+      expect(PlexerJob).to receive(:perform_later).with(druid, version)
+      described_class.perform_now(druid, version)
+    end
+  end
+
+  describe '.zip_binary' do
+    let(:version) { 2 }
+
+    context 'succeeds in zipping the binary' do
+      it 'does not raise an error' do
+        expect { described_class.zip_binary(zip_path, binary_path) }.not_to raise_error
+        File.delete(zip_path)
+      end
+    end
+
+    context 'fails to zip the binary' do
+      context 'when inpath is incorrect' do
```

Comment:
`inpath`?

Reply:
I wasn't sure of what to call the argument, and found `inpath` in the zip manual:

```
Command format.  The basic command format is

              zip options archive inpath inpath ...

       where  archive  is a new or existing zip archive and inpath is a direc-
       tory or file path optionally including wildcards.
```